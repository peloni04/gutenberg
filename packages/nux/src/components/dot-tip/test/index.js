/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import { DotTip } from '..';

jest.mock( '../../../../../components/src/button' );

describe( 'DotTip', () => {
	it( 'should not render anything if invisible', () => {
		const wrapper = shallow(
			<DotTip>
				It looks like you’re writing a letter. Would you like help?
			</DotTip>
		);
		expect( wrapper.isEmptyRender() ).toBe( true );
	} );

	it( 'should render the tip when non-collapsible', () => {
		const wrapper = shallow(
			<DotTip isVisible>
				It looks like you’re writing a letter. Would you like help?
			</DotTip>
		);
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should render a button when collapsible', () => {
		const wrapper = shallow(
			<DotTip isCollapsible isVisible>
				It looks like you’re writing a letter. Would you like help?
			</DotTip>
		);
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should render a custom label when collapsible', () => {
		const wrapper = shallow(
			<DotTip isCollapsible label="Writing a Letter" isVisible>
				It looks like you’re writing a letter. Would you like help?
			</DotTip>
		);
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should render the tip after being clicked when collapsible', () => {
		const wrapper = shallow(
			<DotTip isCollapsible isVisible>
				It looks like you’re writing a letter. Would you like help?
			</DotTip>
		);

		const stopPropagation = jest.fn();
		wrapper.simulate( 'click', { stopPropagation } );
		wrapper.update();

		expect( wrapper ).toMatchSnapshot();
		expect( stopPropagation ).toHaveBeenCalled();
	} );

	it( 'should call onDismiss when the dismiss button is clicked', () => {
		const onDismiss = jest.fn();
		const wrapper = shallow(
			<DotTip isVisible onDismiss={ onDismiss }>
				It looks like you’re writing a letter. Would you like help?
			</DotTip>
		);
		wrapper.find( 'Button[children="Got it"]' ).first().simulate( 'click' );
		expect( onDismiss ).toHaveBeenCalled();
	} );

	it( 'should call onDisable when the X button is clicked', () => {
		const onDisable = jest.fn();
		const wrapper = shallow(
			<DotTip isVisible onDisable={ onDisable }>
				It looks like you’re writing a letter. Would you like help?
			</DotTip>
		);
		wrapper.find( 'IconButton[label="Disable tips"]' ).first().simulate( 'click' );
		expect( onDisable ).toHaveBeenCalled();
	} );
} );
