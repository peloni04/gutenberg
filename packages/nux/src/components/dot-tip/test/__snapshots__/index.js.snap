// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotTip should render a button when collapsible 1`] = `
<button
  aria-label="Open tip"
  className="nux-dot-tip"
  onClick={[Function]}
/>
`;

exports[`DotTip should render a custom label when collapsible 1`] = `
<button
  aria-label="Open tip for “Writing a Letter”"
  className="nux-dot-tip"
  onClick={[Function]}
/>
`;

exports[`DotTip should render the tip after being clicked when collapsible 1`] = `
<button
  aria-label="Close tip"
  className="nux-dot-tip"
  onClick={[Function]}
>
  <Popover
    aria-label="Editor tips"
    className="nux-dot-tip__popover"
    focusOnMount="container"
    noArrow={true}
    onClick={[Function]}
    position="middle right"
    role="dialog"
  >
    <p>
      It looks like you’re writing a letter. Would you like help?
    </p>
    <p>
      <Button
        isLink={true}
      >
        Got it
      </Button>
    </p>
    <IconButton
      className="nux-dot-tip__disable"
      icon="no-alt"
      label="Disable tips"
    />
  </Popover>
</button>
`;

exports[`DotTip should render the tip when non-collapsible 1`] = `
<div
  className="nux-dot-tip"
>
  <Popover
    aria-label="Editor tips"
    className="nux-dot-tip__popover"
    focusOnMount="container"
    noArrow={true}
    onClick={[Function]}
    position="middle right"
    role="dialog"
  >
    <p>
      It looks like you’re writing a letter. Would you like help?
    </p>
    <p>
      <Button
        isLink={true}
      >
        Got it
      </Button>
    </p>
    <IconButton
      className="nux-dot-tip__disable"
      icon="no-alt"
      label="Disable tips"
    />
  </Popover>
</div>
`;
